### ISSUE TITLE
Spamhaus blacklist.

### ISSUE THREAD

**Issue by aikooo7** [DigitalPlatDev/US.KG-Issues#14]
**Describe the bug, issue, or problem:**  
us.kg is in spamhaus's dbl blacklist, making hard to impossible to send emails.

**URL (if applicable):**  
https://check.spamhaus.org/results?query=us.kg

**Domain Name (if applicable):**  
*.us.kg

**Expected Behavior:**  
I wouldn't expect it to not be blacklisted, but what at least you could try remove it from it.

**Additional Context:**  
I tried to do a ticket myself but they want to talk with the owner of us.kg

**Comment by aikooo7** [comment]
I was able to get more information

```
~ $ dig us.kg

; <<>> DiG 9.16.41 <<>> us.kg
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 9607
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;us.kg.                         IN      A

;; ANSWER SECTION:
us.kg.                  3600    IN      A       192.9.243.240

;; Query time: 256 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Sat Aug 31 19:34:18 WEST 2024
;; MSG SIZE  rcvd: 50
```

Doing a dig in us.kg shows it points to a private it further that is likely the reason of the blacklist:

https://mxtoolbox.com/Problem/Blacklist/Spamhaus-DBL/?page=prob_blacklist&ip=us.kg&link=button&action=blacklist:us.kg&showLogin=1&hidetoc=1&reason=127.0.1.6,%20127.0.1.4,%20127.0.1.5

> No IP queries for DBL, see https://www.spamhaus.org/faqs/domain-blocklist#can-the-dbl-be-used-to-look-up-ip-addresses

At the same, seeing the return codes and crossing with the [spamhaus's table](https://www.spamhaus.org/faqs/domain-blocklist#what-do-the-127-return-codes-mean-in-dbl) shows:

127.0.1.6	botnet C&C domain
127.0.1.4	phish domain
127.0.1.5	malware domain

This might be happening because someone is using us kg domains for nerferious activities and us.kg if begin flagged of bad activiity instead of the proper invidivual?

**Comment by EdwardLab** [comment]
We are aware that some domains are being used for fraudulent activities, phishing, and other illegal actions. While we are blocking accounts that are clearly involved in these activities, the large number of registered domains makes it likely that some involved in criminal activity still exist. If possible, please have them email us at contact@digitalplat.org or contact@nic.us.kg to discuss the matter. We are also working to remove US.KG from the DBL list. Thank you.

**Comment by aikooo7** [comment]
> We are aware that some domains are being used for fraudulent activities, phishing, and other illegal actions. While we are blocking accounts that are clearly involved in these activities, the large number of registered domains makes it likely that some involved in criminal activity still exist. If possible, please have them email us at contact@digitalplat.org or contact@nic.us.kg to discuss the matter. We are also working to remove US.KG from the DBL list. Thank you.

I have done a second ticket but I am pretty sure it was deleted as when I access it it is blank. Please let me know of anything else I can help you with and thanks for doing what you can to fix it.

**Comment by EdwardLab** [comment]
They have not been removed from the list. I think we need to ban accounts that are used for criminal activities on a large scale. They said they will not remove them from the list until there are no more domains involved in criminal violations. :(

**Comment by aikooo7** [comment]
> They have not been removed from the list. I think we need to ban accounts that are used for criminal activities on a large scale. They said they will not remove them from the list until there are no more domains involved in criminal violations. :(

Well if we need to go that route:

First I wonder, if the domains are up means that they weren't reported to you, right? Why would someone else report a domain other than the abuse contact, I wonder.

I really hate to say this because this way the unique way that I could get my domains but maybe GitHub kyc is too lax?

Making a tool to scan registered domains for abusive content is a maybe a good idea?

This was only a thing that went into my mind to fix the issue :)

**Comment by slash09** [comment]
v2ray更新不了订阅怎么办?

**Comment by aikooo7** [comment]
@EdwardLab Do you think asking them to treat us.kg as a tld would work? Then they would blacklist domains but us.kg itself no

**Comment by EdwardLab** [comment]
I’ve been so busy lately that I haven't really been managing US.KG or updating the backend (since I'm in school and also working on developing and managing other projects). They won't take care of it, so the only solution is to clean up spam, abuse, and criminal domains to resolve the issue. Once I have some free time, I'll start cleaning things up and tightening the KYC process.

**Comment by aikooo7** [comment]
> I’ve been so busy lately that I haven't really been managing US.KG or updating the backend (since I'm in school and also working on developing and managing other projects). They won't take care of it, so the only solution is to clean up spam, abuse, and criminal domains to resolve the issue. Once I have some free time, I'll start cleaning things up and tightening the KYC process.

Alright that's totally fine, let me know if you need anything

**Comment by aikooo7** [comment]
This https://github.com/DigitalPlatDev/US.KG/commit/4e159b6841f1d394617d5bc9838c98139f9b29b7 and manual review of domains could fix this issue

**Comment by EdwardLab** [comment]
Yes, I use a simple script to scan domains with open ports 443 and 80 to identify those that might be used for criminal purposes.

**Comment by EdwardLab** [comment]
I'll continue to follow up with them to ensure they are removed from the blacklist.

**Comment by zhuzxmas** [comment]
I found the same issue, my email sent from xxx.us.kg will be marked as junk email or spam email, as the domain us.kg is listed on the blacklist...

**Comment by EdwardLab** [comment]
yes, I submitted a removal request ticket to Spamhaus yesterday. they didn't reply to me yet, I'll update here once they reply ticket, just wait first.

> I found the same issue, my email sent from xxx.us.kg will be marked as junk email or spam email, as the domain us.kg is listed on the blacklist...

**Comment by aikooo7** [comment]
> They have not been removed from the list. I think we need to ban accounts that are used for criminal activities on a large scale. They said they will not remove them from the list until there are no more domains involved in criminal violations. :(

Sidenote: Was this fixed? If not it would be pointless to do a ticket, I guess?

**Comment by EdwardLab** [comment]
Update: After scanning, they found that there are still misused domains. We may need to use a domain scanning tool and ban misuse domains to solve it, but I am not sure what tool to use. I will ask for their advice.

I used a script I wrote to scan, but it didn't work very well

**Comment by aikooo7** [comment]
> Update: After scanning, they found that there are still misused domains. We may need to use a domain scanning tool and ban misuse domains to solve it, but I am not sure what tool to use. I will ask for their advice.
> 
> I used a script I wrote to scan, but it didn't work very well

Thanks for the update!

If it was me, I would ask them for recommendation as you said.

**Comment by EdwardLab** [comment]
Update: They seem unwilling to provide scanning tools, suggestions or even answer my questions, and just suggest that I delete these domains or shut down us.kg service. :(

**Comment by aikooo7** [comment]
> Update: They seem unwilling to provide scanning tools, suggestions or even answer my questions, and just suggest that I delete these domains or shut down us.kg service. :(

You could try talking to cloudns.net to check how they do it? They have free domains and aren't on spamhaus's.

https://check.spamhaus.org/results?query=aikoo7.cloudns.ch

**Comment by EdwardLab** [comment]
Update: They provided the latest recommendation to block DNS server resolutions for domains that include patterns like *.com, *.org, *.us (e.g., phishing sites like foo.example.com.sg.us.kg). Since I've been busy lately, I plan to try it out when I have some free time.

**Comment by aikooo7** [comment]
> Update: They provided the latest recommendation to block DNS server resolutions for domains that include patterns like *.com, *.org, *.us (e.g., phishing sites like foo.example.com.sg.us.kg). Since I've been busy lately, I plan to try it out when I have some free time.

Thanks for the update Edward, good tip

**Comment by zhuzxmas** [comment]
> to block DNS server resolutions for domains that include patterns like *.com, *.org, *.us (e.g., phishing sites like foo.example.com.sg.us.kg)

This is very useful information.
Thank you for the update, Edward !

**Comment by EdwardLab** [comment]
Update:
I don't know what to do now :((
![image](https://github.com/user-attachments/assets/aa4016af-6133-4608-979d-50f2c37584d5)
![image](https://github.com/user-attachments/assets/9e16bd2d-7fc2-4813-8305-a42d5ce2ea99)
I'll try my best to fix the issue, but I'm out of ideas right now

**Comment by EdwardLab** [comment]
btw, I’m planning to block registrations with formats like `example-com.us.kg` in the backend later. As for blocking it on DNS, I haven’t figured out a solution yet.

**Comment by aikooo7** [comment]
> Update:
> I don't know what to do now :((
> ![image](https://github.com/user-attachments/assets/aa4016af-6133-4608-979d-50f2c37584d5)
> ![image](https://github.com/user-attachments/assets/9e16bd2d-7fc2-4813-8305-a42d5ce2ea99)
> I'll try my best to fix the issue, but I'm out of ideas right now

Why not block it at registration time? That would be easier and prevent it

**Comment by hyperscandev** [comment]
What programming language is your web app written in?

**Comment by aikooo7** [comment]
Sorry for being annoying but @EdwardLab any updates?

**Comment by cute-omega** [comment]
> Sorry for being annoying but @EdwardLab any updates?

sorry to disturb, but clearly the situation is being worse. From yesterday, all *.us.kg domains cannot be resolved because the registry of *.kg has set the status of US.KG to "Inactive". The inform is [here](https://github.com/DigitalPlatDev/FreeDomain/blob/main/documents/announcement.md) (and if you are using *.us.kg, you may have received a email from digital plat dev yesterday)
still waiting for latest news...

**Comment by aikooo7** [comment]
> > Sorry for being annoying but @EdwardLab any updates?
> 
> sorry to disturb, but clearly the situation is being worse. From yesterday, all *.us.kg domains cannot be resolved because the registry of *.kg has set the status of US.KG to "Inactive". The inform is [here](https://github.com/DigitalPlatDev/FreeDomain/blob/main/documents/announcement.md) (and if you are using *.us.kg, you may have received a email from digital plat dev yesterday). Although I cannot ensure these two things are relevant.
> still waiting for latest news...

It's working now :)
